[IDE] Make the folding margin keyboard accessible

iainx opened the pull request:
When in the text editor, use Alt-Tab to focus the margins.
Tab, Shift+Tab, Left or Right to select the focus margin (Only folding margin
accepts focus at the moment).
Up and Down selects the fold
Space folds or unfolds the selected fold

Fixes VSTS #515126

iainx commented:
![fold keyboard](https://user-images.githubusercontent.com/1253364/36163375-22cc6e9e-10e1-11e8-9eaa-2a102bc79abd.gif)

slluis commented on main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/FoldMarkerMargin.cs:
```diff
@@ -487,7 +494,97 @@ internal protected override void Draw (Cairo.Context cr, Cairo.Rectangle area, D
+		FoldSegment[] focusSegments;
+		int focusedIndex;
+		protected internal override bool SupportsItemCommands => true;
+        protected internal override bool HandleItemCommand(ItemCommand command)
```
Spaces!

slluis commented:
@mkrueger can you please review?

mkrueger commented on main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextArea.cs:
```diff
@@ -668,37 +668,185 @@ void IMCommit (object sender, Gtk.CommitArgs ca)
 			Document.CommitLineUpdate (Caret.Line);
+		}
+
+        protected override bool OnFocusInEvent (EventFocus evnt)
```
Again - spaces. That terrible bug is fixed in theory - but we need the editorconfig support merged in master for that.

15.6 however works.

iainx replied:
argh...

mkrueger commented on main/src/addins/MonoDevelop.SourceEditor2/Mono.TextEditor/Gui/TextViewMargin.cs:
```diff
@@ -2451,7 +2451,7 @@ protected internal override void MouseReleased (MarginMouseEventArgs args)
 		}
 
-		CodeSegmentPreviewWindow previewWindow = null;
+        CodeSegmentPreviewWindow previewWindow = null;
```
Spaces again.

mkrueger commented:
Overall funny feature I didn't implement it that way - but I think it's a nice addition.

iainx commented:
Fixed spaces and added keyboard focus for the icon margin as well
![margins-focus](https://user-images.githubusercontent.com/1253364/36251956-f5a117de-123a-11e8-816f-a3b5b3c16af7.gif)

iainx commented:
@slluis done

iainx commented:
This has been okayed by QA
